src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,49 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+    Animal.alive.push(this);
+  }
+  hide(val) {
+    if (val === undefined) {
+      this.hidden = true;
+    } else if (val === true || false) {
+      this.hidden = val;
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+  bite(val) {
+    if (val instanceof Herbivore) {
+      if (val.health === 50 && !val.hidden === true) {
+        val.health = 0;
+
+        const herbDelIndex = Animal.alive.indexOf(val);
+
+        Animal.alive.splice(herbDelIndex, 1);
+      }
```

Comment:
```suggestion
    if (val instanceof Herbivore && !val.hidden) {
        val.health -= 50;
}
        const herbDelIndex = Animal.alive.indexOf(val);

        Animal.alive.splice(herbDelIndex, 1);
      }
```

And use filter method instead of indexOf and splice